tests/reference/asr-array_01_decl-39cf894.stderr

```diff
@@ -0,0 +1,5 @@
+semantic error: 'i16[:]' object does not support item assignment
```

Comment:
I think this error is not correct, `i16[:]` does support item assignment.

Reply:
> I think this error is not correct, `i16[:]` does support item assignment.

Yes, agreed.
I'm not sure how to handle this, my change was only intended to catch pure integer and floating types like `i8, i16, i32, i64, f32, f64`. I'll look into this soon.

Reply:
I think this diff does the job - basically we want to check the length of the structure. Like integers are dimensionless -
```
+    bool is_integer = ASR::is_a<ASR::Integer_t>(*type);
+    bool is_dimensionless = false;
+    bool is_real = ASR::is_a<ASR::Real_t>(*type);
+    if(is_integer){
+        ASR::Integer_t *integer = ASR::down_cast<ASR::Integer_t>(type);
+        is_dimensionless = integer->n_dims == 0;
+    }
+    if(is_real){
+        ASR::Real_t *real = ASR::down_cast<ASR::Real_t>(type);
+        is_dimensionless = real->n_dims == 0;
+    }
     return ((ASR::is_a<ASR::Character_t>(*type) || ASR::is_a<ASR::Tuple_t>(*type)
-        || ASR::is_a<ASR::Complex_t>(*type)) || ASR::is_a<ASR::Integer_t>(*type) || ASR::is_a<ASR::Real_t>(*type));
+        || ASR::is_a<ASR::Complex_t>(*type)) || is_dimensionless );
 }
```

Reply:
So let us checkout a failing test case -

```
from numpy import empty
def loop_vec():
    a: f64[5] = empty(5)
    i: i32 = 0
    for i in range(5):
        a[i] = 5.0

loop_vec()

(lp) C:\Users\kunni\lpython>python try.py

(lp) C:\Users\kunni\lpython>src\bin\lpython try.py  # No longer catches the error

```
But as soon as I make the array into a floating point variable -
```
def loop_vec():
    a: f64 = 5.0
    i: i32 = 0
    for i in range(5):
        a[i] = 5.1

loop_vec()

(lp) C:\Users\kunni\lpython>python try.py
Traceback (most recent call last):
  File "C:\Users\kunni\lpython\try.py", line 8, in <module>
    loop_vec()
  File "C:\Users\kunni\lpython\try.py", line 6, in loop_vec
    a[i] = 5.1
TypeError: 'float' object does not support item assignment

(lp) C:\Users\kunni\lpython>src\bin\lpython try.py
semantic error: 'f64' object does not support item assignment
 --> try.py:6:9
  |
6 |         a[i] = 5.1
  |         ^^^^^^^^^^


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that must be fixed).
```

Reply:
cc @certik